client/my-sites/checkout/src/components/cost-overrides-list.tsx

```diff
@@ -320,7 +321,11 @@ export function CouponCostOverride( {
 	const { formStatus } = useFormStatus();
 	const isDisabled = formStatus !== FormStatus.READY;
 
-	if ( ! responseCart.coupon || ! responseCart.coupon_savings_total_integer ) {
+	if (
+		! responseCart.coupon ||
+		! responseCart.coupon_savings_total_integer ||
+		isCouponDisplayHidden()
```

Comment:
This feels almost too generic, maybe rename this to reference the affiliate checkout flow more specifically.

Reply:
Thanks @jjchrisdiehl, I made a few changes. Could you give this another review. For this PR I am focusing on the Affiliate Flow. I am still waiting on some feedback regarding removing the coupon box in all instances when the coupon code is embedded in the URL